### DISCUSSION TITLE
Next Release

### DISCUSSION THREAD

**Discussion by NobinKhan** [ravynsoft/ravynos#323] [General]
Hi I am not a technical person about OS but I love this OS and waiting for the next release.
May I know the time for the next release and the version name, please.

**Comment by mszoek** [comment]
The next release will be 0.4.0 and still part of the "Pygmy Marmoset" series (0.4.X). We don't have a fixed date in mind; we're working towards a certain feature set: https://wiki.ravynos.com/wiki/Main_Page. Most of the remaining work is to get Filer working/integrated, fix some bugs, and integrate a terminal app. I had hoped to have VM GPU support done but may push that to 0.5.x so we can get another release out.

**Reply by kjarex** [reply]
As it is very much related: What is missing for the existing (FreeBSD) video drivers to work? 

Like, do they need to be compiled with a ravynOS specific compiler?
Or do they need to be linked to a ravynOS specific library? Or both?

I've been compiling a video driver for the last 3 days (just thought to give it a try - didn't think it would take that long tbh) but with this information I might be able to just cancel it (and either adjust it already or simply give up).

**Reply by mszoek** [reply]
> As it is very much related: What is missing for the existing (FreeBSD) video drivers to work?
> 
> Like, do they need to be compiled with a ravynOS specific compiler? Or do they need to be linked to a ravynOS specific library? Or both?

To actually use them, they need to be built against our kernel version - but that is it. You can do all the dev & test against any FreeBSD 14 system. We have a fork of the drm-kmod repo from FreeBSD and will contribute new working drivers (if we get any) back upstream.

**Reply by kjarex** [reply]
Sorry for the likely stupid question. But for whatever reason compiling this driver has become such a pita, that I don't want to risk doing all that for "nothing": _building against our kernel version_ would be satisfied if I build it within ravynOS, isn't it? I just want to avoid that if I finally succeed the driver might be somehow incompatible and useless.

**Reply by mszoek** [reply]
Not stupid :) Yes, you just need to build it within ravynOS against the current source tree. We track upstream (`freebsd/main` branch, currently 14) so just making it work there should be fine, but you can certainly double check.

What driver are you working on?

**Reply by kjarex** [reply]
xf86-video-scfb is supposed to work for my vm, and it is in ports. So I naively thought to just build it quickly and without any effort and would be done if I'm lucky. I started 3 days ago, and it feels like I had compiled whole freebsd from source at least 10 times by now (for 3 days it was  compiling without any issues (except that it felt really weird and wrong that one would have to compile really that much!?) but today the issues started (autotools, autom4te, perl, python (which it insists to be version 3.9, but had issues with 3 modules), and worst of them all meson, which I haven't been able to get to find llvm for the last 3 hours 😭). 
I guess I just give up - my ravynOS intentions aren't GUI related anyway

**Reply by kjarex** [reply]
it finally finished - but it seems not to be the type of driver I expected and doesn't solve anything :-)

**Reply by NobinKhan** [reply]
Nothing is impossible if you try hard...
I am mostly amazed that you all trying hard to make it better.
I am a Django developer. I don't know if i can be useful in this work but if you need then I will try my best.
